Ginger/GingerTest/ActionConversion/ActionConversionTest.cs

```diff
@@ -0,0 +1,331 @@
+            gen1.LocateBy = Amdocs.Ginger.Common.UIElement.eLocateBy.ByRelXPath;
+            gen1.LocateValue = "//input[@name='first_name']";
+            gen1.GenElementAction = ActGenElement.eGenElementAction.SendKeys;
+            activity.Acts.Add(gen1);
+
+            ActGenElement gen2 = new ActGenElement();
+            gen2.Active = true;
+            gen2.Description = "Set Value : last_name input";
+            gen2.LocateBy = Amdocs.Ginger.Common.UIElement.eLocateBy.ByName;
+            gen2.LocateValue = "last_name";
+            gen2.GenElementAction = ActGenElement.eGenElementAction.SendKeys;
+            activity.Acts.Add(gen2);
+
+            mBF.AddActivity(activity);
+
+            // add business flow to the solution repository
+            mSolutionRepository.AddRepositoryItem(mBF);
+
+            ActionConversionUtils utils = new ActionConversionUtils();
+            ObservableList<ConvertableActionDetails> lst = utils.GetConvertableActivityActions(mBF.Activities.ToList());
+            foreach (var item in lst)
+            {
+                item.Selected = true;
+            }
+            utils.ConvertToActions(false, mBF, lst, true, string.Empty, true);
+
+            Activity newActivity = mBF.Activities[0];
+            int count = newActivity.Acts.Count();
+
+            ////Assert
+            Assert.AreEqual(count, 4);
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void NoActionConversionToSameActivityPOMActionTest()
+        {
+            mBF = new BusinessFlow() { Name = "TestBFConversion", Active = true };
+
+            Activity activity = new Activity();
+            activity.SelectedForConversion = true;
+            ActUIElement gen1 = new ActUIElement();
+            gen1.Active = true;
+            gen1.Description = "Set Value : first_name input";
+            gen1.LocateBy = Amdocs.Ginger.Common.UIElement.eLocateBy.ByRelXPath;
+            gen1.LocateValue = "//input[@name='first_name']";
+            gen1.ElementType = Amdocs.Ginger.Common.UIElement.eElementType.TextBox;
+            gen1.ElementAction = ActUIElement.eElementAction.SendKeys;
+            gen1.ElementLocateValue = "";
+            activity.Acts.Add(gen1);
+
+            mBF.AddActivity(activity);
+
+            // add business flow to the solution repository
+            mSolutionRepository.AddRepositoryItem(mBF);
+
+            ActionConversionUtils utils = new ActionConversionUtils();
+            ObservableList<ConvertableActionDetails> lst = utils.GetConvertableActivityActions(mBF.Activities.ToList());
+            foreach (var item in lst)
+            {
+                item.Selected = true;
+            }
+            utils.ConvertToActions(false, mBF, lst, true, string.Empty, true);
+
+            Activity newActivity = mBF.Activities[0];
+            int count = newActivity.Acts.Count();
+
+            ////Assert
+            Assert.AreEqual(count, 1);
+        }
```

Comment:
i don't see any Test to check that Converstion is really done correctly on the new action itself,
please add test which have Asserts not only for count but also:
1- check the new action type
2- check the new action locateBy / locateValue
3- check the new action operation and value

Reply:
Done